### PR TITLE
Implement parallel collective support for selection I/O.

### PR DESCRIPTION
**Pull request by fortnern**

### CODE REVIEW

**Comment by jrmainzer** [comment]
Other than the need for a formatting run, no obvious issues.

**Review by jrmainzer — APPROVED** [review]

File: src/H5Dcontig.c [hunk 1]
```diff
@@ -658,7 +658,7 @@ H5D__contig_read(H5D_io_info_t *io_info, const H5D_type_info_t *type_info, hsize
         /* Issue selection I/O call (we can skip the page buffer because we've
          * already verified it won't be used, and the metadata accumulator
          * because this is raw data) Only call funciton if nelmts > 0. */
```
Comment (jhendersonHDF) [inline src/H5Dcontig.c]: Does this header comment need updated in relation to now always calling the function below?

File: src/H5Dcontig.c [hunk 2]
```diff
@@ -705,9 +705,9 @@ H5D__contig_write(H5D_io_info_t *io_info, const H5D_type_info_t *type_info, hsiz
         /* Issue selection I/O call (we can skip the page buffer because we've
          * already verified it won't be used, and the metadata accumulator
          * because this is raw data). Only call funciton if nelmts > 0. */
```
Comment (jhendersonHDF) [inline src/H5Dcontig.c]: Same here about updating the comment?

File: src/H5Dcontig.c (hunk 1 above)
Comment (fortnern) [reply src/H5Dcontig.c]: Oops, thanks, I'll fix that when I run the formatter.

**Pull request merged**